src/data/markdown/translated-guides/en/07 Misc/05 k6 Extensions.md

```diff
@@ -1,4 +1,400 @@
+  familiar with both Go and JavaScript, understand how the k6 Go<->JS bridge works,
+  and maintain a public repository for the extension that keeps up to date with any
+  breaking API changes while xk6 is being stabilized.
+
+<!-- This used to be an h3 (nested under "Getting Started"), but h5s aren't styled -->
+<!-- properly and the right side index doesn't show sections above h2, so this is -->
+<!-- at the same level now. :-/ -->
+## Using xk6 to build a k6 binary
+
+You might have found a neat k6 extension on the [Ecosystem page](/ecosystem) or on
+[GitHub](https://github.com/topics/xk6) and wish to use it in your tests. Great! The
+process is relatively simple and we're working on simplifying it further for end
+users.
+
+You'll first need to setup [Go](https://golang.org/doc/install) and
+[Git](https://git-scm.com/). Make sure that your `$PATH` environment variable is
+updated and that `go version` returns the correct version.
+
+Then install xk6 with:
+
+<CodeGroup labels={[]} lineNumbers={[false]}>
+
+```bash
+$ go install github.com/k6io/xk6/cmd/xk6@latest
+```
+
+</CodeGroup>
+
+And confirm that `which xk6` on Linux/macOS or `where xk6` on Windows returns a
+valid path. Otherwise ensure that `$GOPATH` is correctly defined and that
+`$GOPATH/bin` is added to your `$PATH` environment variable. See the
+[Go documentation](https://golang.org/cmd/go/#hdr-GOPATH_environment_variable) for details.
+
+Once xk6 is installed, building a k6 binary with one or more extensions can be done
+with the following command:
+
+<CodeGroup labels={[]} lineNumbers={[false]}>
+
+```bash
+$ xk6 build v0.33.0 \
+  --with github.com/imiric/xk6-sql \
+  --with github.com/szkiba/xk6-prometheus
+```
+
+</CodeGroup>
+
+This will build a `k6` binary in the current directory based on k6 v0.33.0, bundling
+a JavaScript and an Output extension. Now you can run a script with this binary that
+uses the [`xk6-sql` JS API](https://github.com/imiric/xk6-sql) and the
+[Prometheus output](https://github.com/szkiba/xk6-prometheus).
+
+Note that when running the script we have to specify the binary just built in the
+current directory (`./k6`), as otherwise some other `k6` binary found on the system
+could be executed which might not have the extensions built-in. This is only the case
+on Linux and macOS, as Windows shells will execute the binary in the current
+directory first.
+
+
+## Writing a new extension
+
+The first thing you should do before starting work on a new extension is to confirm
+that a similar extension doesn't already exist for your use case. Take a look at
+the [Ecosystem page](/ecosystem) and the [`xk6` topic on GitHub](https://github.com/topics/xk6).
+For example, if a system you need support for can be tested with a generic protocol
+like MQTT, prefer using [xk6-mqtt](https://github.com/pmalhaire/xk6-mqtt)
+instead of creating an extension that uses some custom protocol.
+
```

Comment:
I would also argue to try to implement the functionality in pure JS, which will in general will be way more supported and likely easier to write and reuse

Reply:
Sure, done in 3316dc2.